test/shared_queue_test.cc

```diff
@@ -100,4 +113,42 @@ TEST(SharedQueue, Concurrent) {
   EXPECT_EQ(first_cnt + second_cnt, 200);
 }
 
+TEST(SharedQueue, DeleteByVmId) {
+  SharedQueue shared_queue(false);
+  auto vm_id_1 = "id_1";
+  auto vm_id_2 = "id_2";
+  std::string_view vm_key = "vm_key";
+  uint32_t context_id = 1;
+
+  for (auto i = 1; i < 10; i++) {
+    EXPECT_EQ(i,
+              shared_queue.registerQueue(vm_id_1, std::to_string(i), context_id, nullptr, vm_key));
+    EXPECT_EQ(i, shared_queue.resolveQueue(vm_id_1, std::to_string(i)));
+    i++;
+    EXPECT_EQ(i,
+              shared_queue.registerQueue(vm_id_2, std::to_string(i), context_id, nullptr, vm_key));
+    EXPECT_EQ(i, shared_queue.resolveQueue(vm_id_2, std::to_string(i)));
+  }
```

Comment:
Could you split this into 2 separate `for` loops? Having `i++` in the middle of the loop doesn't help with readability of this code.